dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -0,0 +1,214 @@
+    }
+
+    [Fact]
+    public void CollectionNameCantBeNullOrEmpty()
+    {
+        HostApplicationBuilder builder = this.CreateHostBuilder();
+
+        Assert.Throws<ArgumentNullException>(() => this.RegisterCollection(builder.Services, ServiceLifetime.Singleton, collectionName: null!, serviceKey: null));
+        Assert.Throws<ArgumentNullException>(() => this.RegisterCollection(builder.Services, ServiceLifetime.Singleton, collectionName: null!, serviceKey: "notNull"));
+        Assert.Throws<ArgumentException>(() => this.RegisterCollection(builder.Services, ServiceLifetime.Singleton, collectionName: "", serviceKey: null));
+        Assert.Throws<ArgumentException>(() => this.RegisterCollection(builder.Services, ServiceLifetime.Singleton, collectionName: "", serviceKey: "notNull"));
+    }
+
+#pragma warning disable CA1000 // Do not declare static members on generic types
+    public static IEnumerable<object?[]> LiftetimesAndKeys()
+#pragma warning restore CA1000 // Do not declare static members on generic types
+    {
+        foreach (ServiceLifetime lifetime in new ServiceLifetime[] { ServiceLifetime.Scoped, ServiceLifetime.Singleton, ServiceLifetime.Transient })
+        {
+            yield return new object?[] { lifetime, null };
+            yield return new object?[] { lifetime, "key" };
+            yield return new object?[] { lifetime, 8 };
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(LiftetimesAndKeys))]
+    public virtual void CanRegisterVectorStore(ServiceLifetime lifetime, object? serviceKey)
+    {
+        HostApplicationBuilder builder = this.CreateHostBuilder(serviceKey);
+
+        this.RegisterVectorStore(builder.Services, lifetime, serviceKey);
+
+        using IHost host = builder.Build();
+        // let's ensure that concrete types are registered
+        Verify<TVectorStore>(host, lifetime, serviceKey);
+        // and the abstraction too
+        Verify<VectorStore>(host, lifetime, serviceKey);
+    }
+
+    [Theory]
+    [MemberData(nameof(LiftetimesAndKeys))]
+    public void CanRegisterCollections(ServiceLifetime lifetime, object? serviceKey)
+    {
+        HostApplicationBuilder builder = this.CreateHostBuilder(serviceKey);
+
+        this.RegisterCollection(builder.Services, lifetime, serviceKey: serviceKey);
+
+        using IHost host = builder.Build();
+        // Let's ensure that concrete types are registered.
+        Verify<TCollection>(host, lifetime, serviceKey);
+        // And the VectorStoreCollection abstraction.
+        Verify<VectorStoreCollection<TKey, TRecord>>(host, lifetime, serviceKey);
+        // And the IVectorSearchable abstraction.
+        Verify<IVectorSearchable<TRecord>>(host, lifetime, serviceKey);
+    }
+
+    [Theory]
+    [MemberData(nameof(LiftetimesAndKeys))]
+    public virtual void CanRegisterConcreteTypeVectorStoreAfterSomeAbstractionHasBeenRegistered(ServiceLifetime lifetime, object? serviceKey)
```

Comment:
I personally very much use snake_case (see filter tests) for test names precisely for this kind of super long test name, which I personally find really hard to read (there's a reason humans put spaces between words 🤣 ).

But definitely not blocking, we're already inconsistent on this - we'll clean up afterwards.